SpatialGDK/Source/SpatialGDKTests/SpatialGDK/EngineClasses/SpatialVirtualWorkerTranslator/SpatialVirtualWorkerTranslatorTest.cpp

```diff
@@ -88,16 +130,179 @@ VIRTUALWORKERTRANSLATOR_TEST(Given_no_mapping_WHEN_it_is_updated_THEN_return_the
+	TestTrue("VirtualWorker 2 is VW_F", Translator->GetPhysicalWorkerForVirtualWorker(2)->Equals("VW_F"));
+
+	TestTrue("There is no mapping for virtual worker 3", Translator->GetPhysicalWorkerForVirtualWorker(3) == nullptr);
+
+	TestTrue("Local virtual worker ID is known", Translator->GetLocalVirtualWorkerId() == 1);
+	TestTrue("Translator with local virtual worker ID is ready.", Translator->IsReady());
+	TestTrue("LBStrategy spy reports the correct virtual worker ID.", LBStrategySpy->GetVirtualWorkerId() == 1);
+
+	return true;
+}
+
+VIRTUALWORKERTRANSLATOR_TEST(GIVEN_have_a_valid_mapping_WHEN_another_valid_mapping_is_received_THEN_update_accordingly)
+{
+	ULBStrategySpy* LBStrategySpy = NewObject<ULBStrategySpy>();
+	TUniquePtr<SpatialVirtualWorkerTranslator> Translator = MakeUnique<SpatialVirtualWorkerTranslator>(LBStrategySpy, "ValidWorkerOne");
+
+	// Create a valid initial mapping.
+	Worker_ComponentData FirstValidData = {};
+	FirstValidData.component_id = SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID;
+	FirstValidData.schema_type = Schema_CreateComponentData();
+	Schema_Object* FirstValidDataObject = Schema_GetComponentDataFields(FirstValidData.schema_type);
+
+	// The mapping only has the following entries:
+	// 	VirtualToPhysicalWorkerMapping.Add(1, "ValidWorkerOne");
+	// 	VirtualToPhysicalWorkerMapping.Add(2, "ValidWorkerTwo");
+	Schema_Object* FirstValidDataFirstEntryObject = Schema_AddObject(FirstValidDataObject, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_MAPPING_ID);
+	Schema_AddUint32(FirstValidDataFirstEntryObject, SpatialConstants::MAPPING_VIRTUAL_WORKER_ID, 1);
+	SpatialGDK::AddStringToSchema(FirstValidDataFirstEntryObject, SpatialConstants::MAPPING_PHYSICAL_WORKER_NAME, "ValidWorkerOne");
+
+	Schema_Object* FirstValidSecondEntryObject = Schema_AddObject(FirstValidDataObject, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_MAPPING_ID);
+	Schema_AddUint32(FirstValidSecondEntryObject, SpatialConstants::MAPPING_VIRTUAL_WORKER_ID, 2);
+	SpatialGDK::AddStringToSchema(FirstValidSecondEntryObject, SpatialConstants::MAPPING_PHYSICAL_WORKER_NAME, "ValidWorkerTwo");
+
+	// Apply valid mapping to the translator.
+	Translator->ApplyVirtualWorkerManagerData(FirstValidDataObject);
+
+	// Create a second initial mapping.
+	Worker_ComponentData SecondValidData = {};
+	SecondValidData.component_id = SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID;
+	SecondValidData.schema_type = Schema_CreateComponentData();
+	Schema_Object* SecondValidDataObject = Schema_GetComponentDataFields(SecondValidData.schema_type);
+
+	// The mapping only has the following entries:
+	// 	VirtualToPhysicalWorkerMapping.Add(1, "ValidWorkerOne");
+	// 	VirtualToPhysicalWorkerMapping.Add(2, "ValidWorkerThree");
+	Schema_Object* SecondValidDataFirstEntryObject = Schema_AddObject(SecondValidDataObject, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_MAPPING_ID);
```

Comment:
This block of 3 lines is repeated in a lot of places now, I think it's worthwhile to make a helper AddMapping which takes a Worker_ComponentData, VirtualWorkerId, and PhysicalWorkerName and adds the mapping.

Reply:
Also, if you do this, you can remove the comment describing the mapping, because the code will be clear enough.